examples/daal/cpp/source/k_nearest_neighbors/bf_knn_dense_batch.cpp

```diff
@@ -0,0 +1,151 @@
+/* file: bf_knn_dense_batch.cpp */
+/*******************************************************************************
+ * Copyright 2014-2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+/*
+!  Content:
+!    C++ example of k-Nearest Neighbor in the batch processing mode.
+!******************************************************************************/
+
+/**
+ * <a name="DAAL-EXAMPLE-CPP-bf_KNN_DENSE_BATCH"></a>
+ * \example bf_knn_dense_batch.cpp
+ */
+
+#include "daal.h"
+#include "service.h"
+#include <cstdio>
+#include <cstdlib>
+
+using namespace std;
+using namespace daal;
+using namespace daal::algorithms;
+using namespace daal::data_management;
+
+/* Input data set parameters */
+string trainDatasetFileName = "k_nearest_neighbors_train.csv";
+string testDatasetFileName = "k_nearest_neighbors_test.csv";
+
+size_t nFeatures = 5;
+size_t nClasses = 5;
+
+bf_knn_classification::training::ResultPtr trainingResult;
+bf_knn_classification::prediction::ResultPtr predictionResult;
+NumericTablePtr testGroundTruth;
+NumericTablePtr testData;
+
+char *kchar;
+
+void trainModel();
+void testModel();
+void printResults();
+
+int main(int argc, char *argv[]) {
+  checkArguments(argc, argv, 2, &trainDatasetFileName, &testDatasetFileName);
+
+  kchar = getenv("K");
```

Comment:
It isn't minimal example. Please fix. At least it violates SDL

Reply:
It was debug variable, I deleted it.